fix a couple small nits

BillWagner opened the pull request:
There were a couple small issues in #3597 They are addressed here.

/cc @agocke

BillWagner committed:
fix a couple small nits

There were a couple small issues in #3597 They are addressed here.

/cc @agocke

rpetrusha commented on docs/csharp/local-functions-vs-lambdas.md:
```diff
@@ -81,8 +81,7 @@ assigned at the `return` statement.
 are always necessary for lambda expressions. If a local function is never
-converted to a delegate and none of the variables captured by a locatl function
-are converted to delegates, the compiler can avoid heap allocations. 
+converted to a delegate and one of the variables captured by the local function are captured by other lambdas or local functions that are converted to delegates, the compiler can avoid heap allocations. 
```
is this one of the variables, or none of the variables, as in the original version. One makes little sense.
There should be a comma after "converted to a delegate"
are captured --> is captured

rpetrusha reviewed (changes requested):
Left a few comments, @BillWagner.

BillWagner committed:
respond to feedback

@rpetrusha I updated based on all your feedback. Note that I didn't change 'is' to 'are' on the last comment, because the other fix from 'one' to 'none' hopefully fixes that number mismatch.

rpetrusha commented:
I've approved so you can merge when ready, @BillWagner, but the verb should still be is, since both none and one are singular.

BillWagner committed:
fix verb number
